Modules/Microsoft365DSC/DSCResources/MSFT_AADRoleSetting/MSFT_AADRoleSetting.psm1

```diff
@@ -0,0 +1,1232 @@
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationSecret,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint
+    )
+
+    Write-Verbose -Message "Getting configuration of Role"
+    $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftGraph' `
+        -InboundParameters $PSBoundParameters `
+        -ProfileName 'beta'
+
+    Select-MgProfile -Name 'beta'
+    $MaximumFunctionCount = 32000
+
+    Write-Verbose -Message "Getting configuration of Role"
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace "MSFT_", ""
+    $CommandName  = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $nullReturn = $null
+
+    function getUserUPN{
+        [CmdletBinding()]
+        param (
+            [Parameter()]
+            [string]
+            $Id
+        )
+        $user = Get-MgUser -UserId $Id
+        return $user
+    }
+
+    #get role
+    [string]$Filter = $null
+    $Filter = "scopeId eq '/' and scopeType eq 'DirectoryRole' and RoleDefinitionId eq '" + $Id + "'"
+    $Policy = Get-MgPolicyRoleManagementPolicyAssignment -Filter $Filter
+    $RoleDefinition = Get-MgRoleManagementDirectoryRoleDefinition -UnifiedRoleDefinitionId $Id
+    #get Policyrule
+    $role = Get-MgPolicyRoleManagementPolicyRule -UnifiedRoleManagementPolicyId $Policy.Policyid
+
+    $Displayname = $RoleDefinition.DisplayName
+    $ActivationMaxDuration = ($role | Where-Object{$_.Id -eq "Expiration_EndUser_Assignment"}).AdditionalProperties.maximumDuration
+    $ActivationReqJustification = (($role | Where-Object{$_.Id -eq "Enablement_EndUser_Assignment"}).AdditionalProperties.enabledRules) -contains "Justification"
+    $ActivationReqTicket = (($role | Where-Object{$_.Id -eq "Enablement_EndUser_Assignment"}).AdditionalProperties.enabledRules) -contains "Ticketing"
+    $ActivationReqMFA = (($role | Where-Object{$_.Id -eq "Enablement_EndUser_Assignment"}).AdditionalProperties.enabledRules) -contains "MultiFactorAuthentication"
+    $ApprovaltoActivate = (($role | Where-Object{$_.Id -eq "Approval_EndUser_Assignment"}).AdditionalProperties.setting.isApprovalRequired)
+    [array]$ActivateApprovers = (($role | Where-Object{$_.Id -eq "Approval_EndUser_Assignment"}).AdditionalProperties.setting.approvalStages.primaryApprovers)
+    [string[]]$ActivateApprover = @()
+    foreach($Item in $ActivateApprovers.id){
+        $ActivateApprover += (getUserUPN -Id $Item).UserPrincipalName
```

Comment:
Since the getUserUPN function is only one line and it is only used on this line: Doesn't it make sense to use the actual statement here instead of creating a function for that?